src/migrations/tasks/20230116114550_twu_opportunities.ts

```diff
@@ -0,0 +1,154 @@
+import { makeDomainLogger } from "back-end/lib/logger";
+import { console as consoleAdapter } from "back-end/lib/logger/adapters";
+import Knex from "knex";
+
+const logger = makeDomainLogger(consoleAdapter, "migrations", "development");
+
+export enum TWUOpportunityStatus {
+  Draft = "DRAFT",
+  UnderReview = "UNDER_REVIEW",
+  Published = "PUBLISHED",
+  EvaluationResourceQuestions = "EVAL_QUESTIONS",
+  EvaluationChallenge = "EVAL_C",
+  Awarded = "AWARDED",
+  Suspended = "SUSPENDED",
+  Canceled = "CANCELED"
+}
+
+export enum TWUOpportunityEvent {
+  Edited = "EDITED",
+  AddendumAdded = "ADDENDUM_ADDED",
+}
+
+enum TWUServiceArea {
+  Developer = "DEVELOPER",
+  DataSpecialist = "DATA_SPECIALIST",
+  ScrumMaster = "SCRUM_MASTER",
+  DevOpsSpecialist = "DEVOPS_SPECIALIST"
+}
+
+// eslint-disable-next-line no-empty
+export async function up(connection: Knex): Promise<void> {
+  await connection.schema.createTable("twuOpportunities", (table) => {
+    table.uuid("id").primary().unique().notNullable();
+    table.timestamp("createdAt").notNullable();
+    table.uuid("createdBy").references("id").inTable("users").notNullable();
+  });
+  logger.info("Created twuOpportunities table.");
+
+  await connection.schema.createTable("twuOpportunityVersions", (table) => {
+    table.uuid("id").primary().unique().notNullable();
+    table.timestamp("createdAt").notNullable();
+    table.uuid("createdBy").references("id").inTable("users").notNullable();
+    table
+      .uuid("opportunity")
+      .references("id")
+      .inTable("twuOpportunities")
+      .notNullable()
+      .onDelete("CASCADE");
+    table.text("title").notNullable();
+    table.text("teaser").notNullable();
+    table.boolean("remoteOk").defaultTo(false).notNullable();
+    table.text("remoteDesc").notNullable();
+    table.text("location").notNullable();
+    table.specificType("mandatorySkills", "text ARRAY").notNullable();
+    table.specificType("optionalSkills", "text ARRAY").notNullable();
+    table.text("description").notNullable();
+    table.timestamp("proposalDeadline").notNullable();
+    table.timestamp("assignmentDate").notNullable();
+    table.integer("questionsWeight").notNullable();
+    table.integer("challengeWeight").notNullable();
+    table.integer("priceWeight").notNullable();
+    table.enu("serviceArea", Object.values(TWUServiceArea)).notNullable();
```

Comment:
Made the decision to make this a column instead of a table, like swuPhaseCapabilities. This limits us to only one service area per TWU opportunity, but after Brad and I discussed, we determined it would be suitable for our needs.